src/custom/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,26 @@
+import { useState } from 'react'
+
+export const useLocalStorage = (key: string, value: any) => {
```

Comment:
Although we do use localStorage in the app (https://github.com/gnosis/cowswap/blob/4fb7795a5c9d71c348433044f620cf34dd7a9b9e/src/state/index.ts#L2) I think it would be better to not to use directly.
Instead use redux.

Probably add another property to the `affiliate` state https://github.com/gnosis/cowswap/blob/4fb7795a5c/src/custom/state/affiliate/actions.ts

Reply:
@alfetopito please check the updates